### PR TITLE
[Dialogs] Remove `buttonFont` API.

### PR DESCRIPTION
**Pull request by romoore**
The `buttonFont` was deprecated and has no internal usage. It is being removed
so clients can customize the button font directly.

Part of #5418

### CODE REVIEW

**Review by jverkoey — CHANGES_REQUESTED** [review]
Please separate the removal of the API from the internal cleanup of usage of the API into two separate PRs. I realize we've measured that the risk of roll-back is low because there is no internal usage, but I'd like to encourage us to keep deletion changes separate from migration changes for all of the typical rationale behind the benefits of smaller commits.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 1]
```diff
@@ -283,23 +292,28 @@ - (void)setAccessoryView:(UIView *)accessoryView {
...
+        finalButtonFont = [finalButtonFont
+            mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                         scaledForDynamicType:self.mdc_adjustsFontForContentSizeCategory];
+      }
+      if (button.enableTitleFontForState) {
+        [button setTitleFont:finalButtonFont forState:UIControlStateNormal];
+      } else {
+        button.titleLabel.font = finalButtonFont;
+      }
+    }
```
Comment (jverkoey) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m:316]: tip: as discussed offline: this logic may be valuable to pull out to its own PR.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 2]
```diff
@@ -283,23 +292,28 @@ - (void)setAccessoryView:(UIView *)accessoryView {
...
-  if (self.mdc_adjustsFontForContentSizeCategory) {
-    if (self.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable) {
-      finalButtonFont = [finalButtonFont
-          mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                       scaledForDynamicType:self.mdc_adjustsFontForContentSizeCategory];
-    }
-  }
   for (MDCButton *button in self.actionManager.buttonsInActionOrder) {
-    [button setTitleFont:finalButtonFont forState:UIControlStateNormal];
+    if (self.mdc_adjustsFontForContentSizeCategory) {
```
Comment (jverkoey) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m:297]: Can you move `if (self.mdc_adjustsFontForContentSizeCategory) {` outside of the loop as a guard statement? E.g.:

```objc
  if (!self.mdc_adjustsFontForContentSizeCategory) {
    return;
  }
```

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 3]
```diff
@@ -129,7 +129,16 @@ - (void)addActionButton:(nonnull MDCButton *)button {
...
       [button setTitleColor:_buttonColor forState:UIControlStateNormal];
     }
-    [button setTitleFont:_buttonFont forState:UIControlStateNormal];
+    if (button.enableTitleFontForState) {
+      if (![button titleFontForState:UIControlStateNormal]) {
+        [button setTitleFont:[MDCAlertControllerView buttonFontDefault]
+                    forState:UIControlStateNormal];
+      }
+    } else {
+      if (!button.titleLabel.font) {
```
Comment (jverkoey) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m:138]: This can be flattened to:

```objc
    } else if (!button.titleLabel.font) {
```

**Review by jverkoey — APPROVED** [review]
Rob and I discussed this PR offline. Summary of the discussion is:

- The changes in this PR represent the result of deleting the API in question and addressing the build failures that were discovered.
- Keeping the API around would require a temporary increase in complexity both in tests and implementation.
- We agreed that that temporary increase in complexity is likely not worth the reduction in PR size / ideas being tackled here.
- There is some new logic that was added in this PR that Rob may decide to pull out to a separate PR or document in this PR's description.

File: components/Dialogs/examples/DialogsTallTextAlertExampleViewController.m [hunk 4]
```diff
@@ -93,7 +94,14 @@ - (IBAction)didTapPresent:(id)sender {
     dialogButtonFont = [UIFont fontWithName:urduFontName size:20.0];
   }
   alert.messageFont = dialogBodyFont;
-  alert.buttonFont = dialogButtonFont;
+  for (MDCAlertAction *action in alert.actions) {
+    MDCButton *button = [alert buttonForAction:action];
+    if (button.enableTitleFontForState) {
```
Comment (galiak11) [inline components/Dialogs/examples/DialogsTallTextAlertExampleViewController.m:99]: Is enableTitleFontForState true by default? Why are there 2 ways to set font for the button?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 3 above)
Comment (galiak11) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:138]: Why are we checking this instead of simply overriding the existing font, like in the other places?

**Review by galiak11 — APPROVED** [review]

**Comment by jverkoey** [comment]
Closing this PR due to inactivity, but please re-open it again if / when you intend to pursue it.
